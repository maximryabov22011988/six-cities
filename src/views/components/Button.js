import React from 'react';
import PropTypes from 'prop-types';
import cn from 'classnames';

const propTypes = {
  className: PropTypes.string.isRequired,
  disabled: PropTypes.bool,
  type: PropTypes.string,
};
const defaultProps = {
  type: 'button',
};

function Button({ className, type, disabled, children }) {
  return (
    <button className={cn(className, 'button')} disabled={disabled} type={type}>
      {children}
    </button>
  );
}

Button.propTypes = propTypes;
Button.defaultProps = defaultProps;

export default Button;
