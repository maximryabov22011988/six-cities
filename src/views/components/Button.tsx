import * as React from 'react';
import cn from 'classnames';

interface Props {
  children: React.ReactNode,
  className: string,
  disabled?: boolean,
  type?: 'button' | 'reset' | 'submit',
  onClick?: (event: React.MouseEvent<HTMLButtonElement>) => void,
}

function Button({ className, type = 'button', disabled, children, onClick }: Props) {
  return (
    <button className={cn(className, 'button')} disabled={disabled} type={type} onClick={onClick}>
      {children}
    </button>
  );
}

export default Button;
