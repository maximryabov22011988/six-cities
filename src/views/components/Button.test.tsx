import * as React from 'react';
import * as renderer from 'react-test-renderer';

import Button from './Button';

it('Button correctly render', () => {
  const tree = renderer
    .create(
      <Button className="login__submit form__submit">Button</Button>
    ).toJSON();

  expect(tree).toMatchSnapshot();
});
