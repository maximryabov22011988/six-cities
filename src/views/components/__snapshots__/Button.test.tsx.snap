// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button correctly render 1`] = `
<button
  className="login__submit form__submit button"
  type="button"
>
  Button
</button>
`;
